force-app/main/default/classes/LeadTriggerHandlerTest.cls

```diff
@@ -0,0 +1,163 @@
+                } else if (ld.Title.containsIgnoreCase('exec') || ld.Title.equalsIgnoreCase('chief') || ld.Title.containsIgnoreCase('head')) {
+                    Assert.areEqual('Executive', ld.Title, 'Title should be normalized to Executive');
+                } else if (ld.Title.containsIgnoreCase('assist') || ld.Title.containsIgnoreCase('deputy') || ld.Title.equalsIgnoreCase('jr')) {
+                    Assert.areEqual('Assistant', ld.Title, 'Title should be normalized to Assistant');
+                }
+            }
+        }
+    }
+
+    @isTest 
+    static void test_handleAutoLeadScoring() {
+        // List of industries and lead sources to randomly select
+        List<String> listOfIndustries = new List<String>{'Technology', 'Agriculture', 'Chemicals', 'Banking', 'Education'};
+        List<String> listOfLeadSources = new List<String>{'Web', 'Purchased List','Phone Inquiry', 'Partner Referral'};
+
+        List<Lead> leadsToInsert = new List<Lead>();
+        
+        // Generate a random indistry and lead source to use for all leads
+        Integer randomIndustryIndex = (Integer) Math.floor(Math.random() * listOfIndustries.size());
+        Integer randomSourceIndex = (Integer) Math.floor(Math.random() * listOfLeadSources.size());
+
+        String randomIndustry = listOfIndustries[randomIndustryIndex];
+        String randomSource = listOfLeadSources[randomSourceIndex];
+        
+        Integer totalLeadScore = 18;
+
+        for (Integer i = 0; i < 400; i++) {
+            String formattedIndex = padLeft(String.valueOf(i), 3);
+            if (Math.mod(i, 2) == 0) {
+                leadsToInsert.add(new Lead(FirstName = 'Test', LastName = 'Lead ' + i, Company = 'Test Company' + formattedIndex, Industry = randomIndustry, LeadSource = randomSource, Phone = '123-456-' + formattedIndex));
+            } else {
+                leadsToInsert.add(new Lead(FirstName = 'Test', LastName = 'Lead ' + i, Company = 'Test Company' + formattedIndex, Industry = randomIndustry, LeadSource = randomSource, Email = 'Test' + formattedIndex + '@sfmail.com'));
+            }
+        }
+
+        Test.startTest();
+        insert leadsToInsert;
+        Test.stopTest();
+
+        List<Lead> insertedLeads = [SELECT Id, FirstName, LastName, Industry, LeadSource, Lead_Score__c, Email, Phone FROM Lead WHERE Id IN :leadsToInsert];
+        for (Lead ld : insertedLeads) {
+            if (ld.LeadSource == 'Web' && ld.Email != null) {
+				Assert.areEqual(3, 3, 'Score should at least be 3');
+			}
+
+			if (ld.Phone != null) {
+				Assert.areEqual(5, 5, 'Score should at least be 5');
+			}
+
+			if (ld.Industry == 'Technology') {
+				Assert.areEqual(10, 10, 'Score should at least be 10');
+			}
+
+			if (totalLeadScore > 18) {
+				Assert.areEqual(18, 18, 'Score should not exceed 18');
+			}
+        }
+    }
+
+    public static String padLeft(String originalString, Integer targetLength) {
```

Comment:
This can probably be a private method if its only being used by other methods in this class